### PR TITLE
[DBC-226] Implement permission guide

### PR DESCRIPTION
**Pull request by quocnd**
## Description
- Implement logic in permission guide

## Note
- Need to update text and icons for permissions once we have final design

## Screenshots
https://drive.google.com/file/d/1_IFc3_nYISSfljArZfxSrG36XMaqi3oQ/view?usp=sharing

image

### CODE REVIEW

**Review by cutehackers — CHANGES_REQUESTED** [review]
- GlobalParam. isPermissionGuideShown is not such a good idea to cache the value because it's already store in preference. But it's ok if it's a temporary use. otherwise you are required to refactor the code. 
- PermissionGuideFragment screen is need to have an argument that makes it navigate to authGuide screen, not from GlobalParam.

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/LauncherViewModel.kt [hunk 1]
```diff
@@ -15,9 +15,34 @@
...
+    private val getPermissionStateUseCase: GetPermissionStateUseCase,
+) : BaseViewModel() {
+
+    fun getPermissionState() {
+        viewModelScope.launch {
+            getPermissionStateUseCase.invoke(Any())
+                .collect {
+                    when (it) {
+                        is Resource.Success -> {
+                            GlobalParam.isPermissionGuideShown = it.data
```
[the remark this answers is not in the document]
Comment (quocnd) [reply app/src/main/java/io/mvlchain/tada/wallet/app/ui/LauncherViewModel.kt]: I got an issue with load time with previous test.
It took more than 1s to get `PermissionGuideState`. It meant if I pass via intent it always false.
I just retest and can't reproduce this issue now. 😢

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/permissionguide/PermissionGuideFragment.kt [hunk 2]
```diff
@@ -15,28 +15,46 @@
...
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        initEvents()
+    override fun getBindingInflater(): BindingInflater<FragmentPermissionGuideBinding> {
+        return FragmentPermissionGuideBinding::inflate
     }
 
-    private fun initEvents() {
+    override fun initViews() {
+        if (GlobalParam.isPermissionGuideShown) {
```
[the remark this answers is not in the document]
Comment (quocnd) [reply app/src/main/java/io/mvlchain/tada/wallet/app/ui/permissionguide/PermissionGuideFragment.kt]: With above reason, but now it's fixed

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/LauncherActivity.kt [hunk 3]
```diff
@@ -28,19 +29,28 @@ import kotlinx.coroutines.launch
 @AndroidEntryPoint
 class LauncherActivity : AppCompatActivity() {
 
+    companion object {
+        const val EXTRA_PERMISSION_GUIDE_SHOWN = "ExtraPermissionGuideState"
```
Comment (cutehackers) [inline app/src/main/java/io/mvlchain/tada/wallet/app/ui/LauncherActivity.kt:33]: Please use the same values as its name

**Review by cutehackers — APPROVED** [review]

**Pull request merged**